src/plugins/intel_gna/src/optimizer/gna_pass_manager.cpp

```diff
@@ -2436,6 +2506,85 @@ void TransposeWeightsFromNCHWToNHWCPass::run() {
     }
 }
 
+void FuseFullyConnectedWithEltwisePass::run() {
+    // This legacy pass removes the Eltwise (only if it performs SUM op) from between FC and Any.
+    // The Const layer blob data is added to biases blob data of FC layer.
+    // Finally Const and Reshape (if exists) are also removed
+    //
+    // This operation can be illustrated as follows:
+    //
+    // --
+    //  Original:               Result:         Removed:
+    //
+    //    FC                     FC              (Eltwise)
+    //    |                      |               (Const)
+    //  Reshape                 Any              (Reshape)
+    // (optional)  Const
+    //    |       /
+    //  Eltwise(sum)
+    //    |
+    //  Any (e.g. ReLU)
+    //--
+    //
+    // NOTE: This pass is implemented to prevent unnecessary roundtrip to memory (additional layer).
+    // It can be fully removed if corresponding LPT transformation is already implemented.
+
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "FuseFullyConnectedWithEltwisePass");
+    for (auto& l : *pLayers) {
+        if (!LayerInfo(l).isFullyConnected() || !l->outData.size()) {
```

Comment:
Is it possible to have a FC layer with empty outData? It should raise an exception much earlier, I guess. I don't think it makes sense to check this here.

Reply:
Condition removed.